stdlib/rubygems/0/version.rbs

```diff
@@ -1,3 +1,228 @@
+  # are worried about incompatible future changes (it is OK to be pessimistic!).
+  #
+  # ## Preventing Version Catastrophe:
+  #
+  # From: http://blog.zenspider.com/2008/10/rubygems-howto-preventing-cata.html
+  #
+  # Let's say you're depending on the fnord gem version 2.y.z. If you specify your
+  # dependency as ">= 2.0.0" then, you're good, right? What happens if fnord 3.0
+  # comes out and it isn't backwards compatible with 2.y.z? Your stuff will break
+  # as a result of using ">=". The better route is to specify your dependency with
+  # an "approximate" version specifier ("~>"). They're a tad confusing, so here is
+  # how the dependency specifiers work:
+  #
+  #     Specification From  ... To (exclusive)
+  #     ">= 3.0"      3.0   ... &infin;
+  #     "~> 3.0"      3.0   ... 4.0
+  #     "~> 3.0.0"    3.0.0 ... 3.1
+  #     "~> 3.5"      3.5   ... 4.0
+  #     "~> 3.5.0"    3.5.0 ... 3.6
+  #     "~> 3"        3.0   ... 4.0
+  #
+  # For the last example, single-digit versions are automatically extended with a
+  # zero to give a sensible result.
+  #
+  class Version
+    include Comparable
+
+    # True if the `version` string matches RubyGems' requirements.
+    #
+    def self.correct?: (_ToS version) -> bool
+
+    # Factory method to create a Version object. Input may be a Version or a String.
+    # Intended to simplify client code.
+    #
+    #     ver1 = Version.create('1.3.17')   # -> (Version object)
+    #     ver2 = Version.create(ver1)       # -> (ver1)
+    #     ver3 = Version.create(nil)        # -> nil
+    #
+    def self.create: (_ToS | Version | nil input) -> instance?
+
+    # Constructs a Version from the `version` string.  A version string is a series
+    # of digits or ASCII letters separated by dots.
+    #
+    def initialize: (_ToS version) -> void
+
+    # is larger, the same, or smaller than this one. Attempts to compare to
+    # something that's not a `Gem::Version` return `nil`.
+    #
+    def <=>: (untyped other) -> Integer?
+
+    # A recommended version for use with a ~> Requirement.
+    #
+    def approximate_recommendation: () -> untyped
+
+    # Return a new version object where the next to the last revision number is one
+    # greater (e.g., 5.3.1 => 5.4).
+    #
+    # Pre-release (alpha) parts, e.g, 5.3.1.b.2 => 5.4, are ignored.
+    #
+    def bump: () -> instance
+
+    def canonical_segments: () -> Array[Integer | String]
+
+    # A Version is only eql? to another version if it's specified to the same
+    # precision. Version "1.0" is not the same as version "1".
+    #
+    def eql?: (untyped other) -> bool
+
+    # Dump only the raw version string, not the complete object. It's a string for
+    # backwards (RubyGems 1.3.5 and earlier) compatibility.
+    #
+    def marshal_dump: () -> Array[String]
+
+    # Load custom marshal format. It's a string for backwards (RubyGems 1.3.5 and
+    # earlier) compatibility.
+    #
+    def marshal_load: (Array[String] array) -> void
```

Comment:
[note] `#marshal_load` returns `nil` actually, but I think it should return `void` because `#marshal_load` calls internally `#initialize`:

https://github.com/rubygems/rubygems/blob/v3.2.11/lib/rubygems/version.rb#L273-L275